src/components/FormikAutocomplete.jsx

```diff
@@ -5,16 +5,28 @@ import PropTypes from 'prop-types';
 
 /* eslint-disable react/jsx-props-no-spreading */
 function FormikAutocomplete(props) {
-  const { label, id, labelStyle, formik, placeholder, ...otherProps } = props;
+  const { label, id, labelStyle, formik, placeholder, options, ...otherProps } =
+    props;
   const values = getIn(formik.values, id, []);
+
+  const formatTags = (vals) => {
+    const newVals = new Set(
+      vals.map((v) => v.toLowerCase().replace(/\s+/g, '-'))
+    );
+    formik.setFieldValue(id, Array.from(newVals));
+  };
+
   return (
     <>
       <InputLabel sx={labelStyle}>{label}</InputLabel>
       <Autocomplete
         id={id}
         multiple
         value={values}
-        onChange={(e, val) => formik.setFieldValue(id, val)}
+        options={options}
+        onChange={(e, val) =>
+          id === 'tags' ? formatTags(val) : formik.setFieldValue(id, val)
+        }
```

Comment:
This makes the component a bit fragile as handling other special use cases like this would only complicate it more. Instead of this how about just accepting an optional `onChange` property? Then this can look like:
```suggestion
        onChange={onChange ?? (e, val) => formik.setFieldValue(id, val)}
```

Reply:
did it slightly different